doc/rst/developer/chips/23.rst

```diff
@@ -0,0 +1,614 @@
+   unappealing.
+
+   The role of super.init() as the separator between phase 1 and
+   phase 2 remains unclear to some users.
+
+   Chapel allows a developer to omit an explicit call to a parent
+   initializer and rely on the compiler to insert this call.  It
+   is unclear whether it is more natural for an initalizer to
+   default to phase 1 or phase 2, i.e. whether the compiler
+   should insert the call at the start of the body or at the end
+   of the body, and it is hard to recall which choice has been made
+   for Chapel.
+
+   The Chapel language has been modified to disallow the potential to
+   support inheritance for records but we continue to rely on a call
+   to super.init() to signal that the type has been fully initialized.
+
+
+
+
+
+
+
+
+
+
+Additional concerns
++++++++++++++++++++
+
+3. The current implementation suppresses the compiler generated
+   initializers if the developer defines at least one
+   initializer. This ensures that a user does not unintentionally
+   rely on a compiler-generated all-fields initializer that fails
+   to meet the necessary semantics, but may require the type designer
+   to explicity write this initializer even in the simple cases.
+
+   The type designer also becomes responsible for the default
+   copy initializer if the type is a record.
+
+
+
+
+4. It may be unclear how to define a type specifier for a generic
+   type or how to define a minimal initializer.  For example what
+   is the signature for the compiler generated type constructor
+   in the following code fragment
+
+
+   .. code-block:: chapel
+
+      record MyR {
+        param flag  : bool;
+        var   value : int;
+
+        <a user defined initializer>
+      }
+
+      var myR : MyR( ... );
+
+
+   and what is the signature for a user-defined initializer that
+   will satisfy the variable declaration?  Should the answer be
+   different if this were a class instead of a record?
+
+   The compiler implements defensible answers to these questions
+   but it is easy to be momentarily surprised by the answers.
+   Perhaps more so for the user-defined initializer than the
+   compiler generated type constructor.
+
+
+
+
+5. We continue to struggle with copy initializers.  A copy
+   initializer accepts one formal with the same type as the
+   record.  It is not always clear when the compiler will
+   attempt to use a copy initializer or how the user can
+   unambiguously define the copy initializer in the presence
+   of other initializers that accept a single formal.
+
+
+
+
+
+
+
+
+
+
+
+Initialization vs. assignment
++++++++++++++++++++++++++++++
+
+It is important to distinguish between initialization and assignment
+for fields with record type.  Mutable objects must be initialized once
+and may be assigned any number of times after that.  Failing to
+support this requirement correctly may have consequences for
+correctness or performance. Consider the following contrived example
```

Comment:
Ditto here.  I don't follow what bearing this has on the chip, or a difference relative to or wrt what's in master.   But ditto also the part about ignoring me.